contrib/epee/src/mlog.cpp

```diff
@@ -134,9 +135,45 @@ void mlog_configure(const std::string &filename_base, bool console, const std::s
   el::Loggers::addFlag(el::LoggingFlag::DisableApplicationAbortOnFatalLog);
   el::Loggers::addFlag(el::LoggingFlag::ColoredTerminalOutput);
   el::Loggers::addFlag(el::LoggingFlag::StrictLogFileSizeCheck);
-  el::Helpers::installPreRollOutCallback([filename_base](const char *name, size_t){
+  el::Helpers::installPreRollOutCallback([filename_base, max_log_files](const char *name, size_t){
     std::string rname = generate_log_filename(filename_base.c_str());
     rename(name, rname.c_str());
+    if (max_log_files != 0)
+    {
+      std::vector<boost::filesystem::path> found_files;
+      const boost::filesystem::directory_iterator end_itr;
+      for (boost::filesystem::directory_iterator iter(boost::filesystem::path(filename_base).parent_path()); iter != end_itr; ++iter)
+      {
+        const std::string filename = iter->path().string();
+        if (filename.size() >= filename_base.size() && filename.substr(0, filename_base.size()) == filename_base)
+        {
+          found_files.push_back(iter->path());
+        }
+      }
+      if (found_files.size() > max_log_files)
+      {
+        std::sort(found_files.begin(), found_files.end(), [](const boost::filesystem::path &a, const boost::filesystem::path &b) {
+          time_t ta, tb;
+          if (file_io_utils::get_file_time(a.string(), ta) && file_io_utils::get_file_time(b.string(), tb))
+          {
+            return ta < tb;
+          }
+          return false;
+        });
+        for (size_t i = 0; i < found_files.size() - max_log_files; ++i)
+        {
+          try
+          {
+            boost::system::error_code ec{};
+            boost::filesystem::remove(found_files[i], ec);
+          }
+          catch (const std::exception &e)
+          {
+            MERROR("Failed to remove " << found_files[i] << ": " << e.what());
+          }
```

Comment:
If you switch to the ec version, it doesn't throw, it's one xor the other :)

Reply:
Ah, I was too lazy to check the reference and simply copied a code snippet from common/util.cpp :)